src/configuration/raw.rs

```diff
@@ -0,0 +1,155 @@
+//! internal implementation details for handling configuration XML
+
+use std::{path::PathBuf, str::FromStr};
+
+use serde::Deserialize;
+
+use super::{
+    policy::{RuleMatch, RuleMatchType},
+    ApparmorMode, Associate, IncludeElement, LimitElement, Principal, Type,
+};
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub(super) struct RawApparmor {
+    #[serde(rename = "@mode")]
+    pub mode: Option<ApparmorMode>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
+#[serde(default)]
+pub(super) struct RawConfiguration {
```

Comment:
It's under `raw` mod, so `Raw` prefixes in the names are unnecessary. If your'e coming from a C background, I know from my personal experience, that this is one habit that's hard to get rid of. :)

Reply:
Some more stuff:

* Instead of `pub(super) everywhere, you can just have them `pub` and then from the supermodule, you keep the mod private and that achieves the same.
* It would be good to have some docs for each time ideally but one thing that definitely need some docs is what the raw API is all about (through a module-level documentation comment).